sdk/core/azure_core_amqp/src/error.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corp. All Rights Reserved.
+// cspell: words amqp
+
+#[cfg(all(feature = "enable-fe2o3-amqp", not(target_arch = "wasm32")))]
+use crate::fe2o3::error::Fe2o3AmqpError;
+
+pub enum ErrorKind {
+    #[cfg(all(feature = "enable-fe2o3-amqp", not(target_arch = "wasm32")))]
+    IronOxideError { source: Fe2o3AmqpError },
+    #[cfg(any(not(feature = "enable-fe2o3-amqp"), target_arch = "wasm32"))]
+    NoopError,
+}
+
+pub struct AmqpError {
+    kind: ErrorKind,
+}
+
+impl AmqpError {
+    #[cfg(all(feature = "enable-fe2o3-amqp", not(target_arch = "wasm32")))]
+    pub fn new_iron_oxide_error(source: Fe2o3AmqpError) -> Self {
+        Self {
+            kind: ErrorKind::IronOxideError { source },
+        }
+    }
+}
+impl std::error::Error for AmqpError {
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+        None
```

Comment:
I assume this is just temporary since you actually stored the underlying error as `source`?

Reply:
No, it was a cut&paste thingy I picked up from somewhere, not even sure where (could have been copilot halucinating, tbh).

Its fixed now.

Reply:
Fixed in both AMQP and Fe2o3.